src/commands/utility/checksocialcredit.ts

```diff
@@ -17,15 +18,16 @@ abstract class checkSocialCredit extends GenericCommand {
 	}
 
 	async run(message: Message, args: string[], addCD: () => Promise<void>) {
-		const member = await resolveMember(args[0], message.guild!).catch(() => message.member);
+		if (!message.guild) throw new UserError({ identifier: Identifiers.PreconditionGuildOnly }, 'guild only command');
+		const member = await resolveMember(args[0], message.guild).catch(() => message.member);
+		if (!member) throw new Error('missing member');
+		const user = (await getUser(member.id)) ?? (await createUser(member.id));
+		if (!user) throw new Error('missing user database entry');
```

Comment:
this is a good example of code that could be refactored into methods. `getOrCreate{..}` and `get{..}` would be cleaner methods and I would place the error throwing in those methods.